FWCore/Concurrency/interface/FinalWaitingTask.h

```diff
@@ -0,0 +1,60 @@
+#ifndef FWCore_Concurrency_FinalWaitingTask_h
+#define FWCore_Concurrency_FinalWaitingTask_h
+// -*- C++ -*-
+//
+// Package:     FWCore/Concurrency
+// Class  :     FinalWaitingTask
+//
+/**\class FinalWaitingTask FinalWaitingTask.h "FWCore/Concurrency/interface/FinalWaitingTask.h"
+
+ Description: [one line class summary]
+
+ Usage:
+   Use this class on the stack to signal the final task to be run.
+   Call done() to check to see if the task was run and check value of
+   exceptionPtr() to see if an exception was thrown by any task in the group.
+
+*/
+//
+// Original Author:  Christopher Jones
+//         Created:  Tue, 12 Jul 2022 18:45:15 GMT
+//
+
+// system include files
+#include "oneapi/tbb/task_group.h"
+
+// user include files
+#include "FWCore/Concurrency/interface/WaitingTask.h"
+
+// forward declarations
+namespace edm {
+  class FinalWaitingTask : public WaitingTask {
+  public:
+    FinalWaitingTask() = delete;
+    explicit FinalWaitingTask(tbb::task_group& iGroup)
+        : m_group{&iGroup}, m_handle{iGroup.defer([]() {})}, m_done{false} {}
+
+    void execute() final { m_done = true; }
+
+    [[nodiscard]] bool done() const noexcept { return m_done.load(); }
+
+    void wait() {
```

Comment:
Do we care about having similar nomenclature with SYCL queues? (asking since SYCL/DPC++ is part of the oneAPI) There the `wait()` does not throw, and one has to ask `wait_and_throw()` explicitly to have exceptions (from asynchronous processing) to throw.

Reply:
I don't think we care at all. I went with consistency with `tbb::task_group::wait` which can throw.

Reply:
Right, consistency with oneTBB's nomenclature is more important.